core/src/main/java/com/graphhopper/routing/GraphHopperRouter.java

```diff
@@ -0,0 +1,312 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.routing;
+
+import com.graphhopper.GHRequest;
+import com.graphhopper.GHResponse;
+import com.graphhopper.config.LMProfile;
+import com.graphhopper.config.Profile;
+import com.graphhopper.routing.ch.CHPreparationHandler;
+import com.graphhopper.routing.ch.CHRoutingAlgorithmFactory;
+import com.graphhopper.routing.lm.LMPreparationHandler;
+import com.graphhopper.routing.querygraph.QueryGraph;
+import com.graphhopper.routing.template.AlternativeRoutingTemplate;
+import com.graphhopper.routing.template.RoundTripRoutingTemplate;
+import com.graphhopper.routing.template.RoutingTemplate;
+import com.graphhopper.routing.template.ViaRoutingTemplate;
+import com.graphhopper.routing.util.DefaultEdgeFilter;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.TraversalMode;
+import com.graphhopper.routing.weighting.BlockAreaWeighting;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.storage.CHConfig;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.GraphEdgeIdFinder;
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.storage.index.LocationIndex;
+import com.graphhopper.storage.index.QueryResult;
+import com.graphhopper.util.*;
+import com.graphhopper.util.details.PathDetailsBuilderFactory;
+import com.graphhopper.util.exceptions.PointDistanceExceededException;
+import com.graphhopper.util.exceptions.PointOutOfBoundsException;
+import com.graphhopper.util.shapes.BBox;
+import com.graphhopper.util.shapes.GHPoint;
+
+import java.util.*;
+
+import static com.graphhopper.routing.weighting.Weighting.INFINITE_U_TURN_COSTS;
+import static com.graphhopper.util.Helper.DIST_EARTH;
+import static com.graphhopper.util.Parameters.Algorithms.*;
+import static com.graphhopper.util.Parameters.Routing.CURBSIDE;
+import static com.graphhopper.util.Parameters.Routing.POINT_HINT;
+
+public class GraphHopperRouter {
+    private final GraphHopperStorage ghStorage;
+    private final EncodingManager encodingManager;
+    private final LocationIndex locationIndex;
+    private final Map<String, Profile> profilesByName;
+    private final PathDetailsBuilderFactory pathDetailsBuilderFactory;
+    private final TranslationMap translationMap;
+    private final RoutingConfig routingConfig;
+    private final WeightingFactory weightingFactory;
+    private final CHPreparationHandler chPreparationHandler;
+    private final LMPreparationHandler lmPreparationHandler;
```

Comment:
Similar to `GraphHopper` these `PreparationHandlers` are concerned with both the 'import' and the 'algorithm creation' (when routing), but they are mostly about preparation and I definitely want to get rid of them here.